src/codemagic/models/certificate.py

```diff
@@ -52,6 +52,12 @@ def from_ans1(cls, asn1: AnyStr) -> Certificate:
         x509_certificate = cls._get_x509_certificate(asn1, crypto.FILETYPE_ASN1)
         return Certificate(x509_certificate)
 
+    @classmethod
+    def from_p12(cls, p12: AnyStr, password: Optional[AnyStr] = None) -> Certificate:
+        p12_archive = crypto.load_pkcs12(p12, password)
```

Comment:
`crypto.load_pkcs12` emits warnings if str passwords are given.
```suggestion
        passphrase = cls._bytes(password) if password else None
        p12_archive = crypto.load_pkcs12(p12, passphrase)
```